packages/frontend/src/components/PieChartConfig/PieChartLayoutConfig.tsx

```diff
@@ -0,0 +1,135 @@
+        customMetrics,
+        tableCalculations,
+        pieChartConfig: {
+            metricId,
+            metricChange,
+
+            groupFieldIds,
+            groupAdd,
+            groupChange,
+            groupRemove,
+
+            // isDonut,
+            // isDonutChange,
+        },
+    } = useVisualizationContext();
+
+    const selectedMetric = useMemo(() => {
+        return [...metrics, ...customMetrics, ...tableCalculations].find((m) =>
+            isField(m) ? fieldId(m) === metricId : m.name === metricId,
+        );
+    }, [metrics, customMetrics, tableCalculations, metricId]);
+
+    return (
+        <Stack w={320}>
+            <Stack spacing="xs">
+                {groupFieldIds.map((dimensionId, index) => {
+                    const dimension = dimensions.find(
+                        (d) => fieldId(d) === dimensionId,
+                    );
+
+                    return (
+                        <Select
+                            key={index}
+                            clearable={index !== 0}
+                            label={index === 0 ? 'Group' : undefined}
+                            placeholder="Select dimension"
+                            icon={dimension && <FieldIcon item={dimension} />}
+                            value={dimensionId}
+                            data={dimensions.map((d) => ({
+                                item: d,
+                                label: fieldLabelText(d),
+                                value: fieldId(d),
+                                disabled: groupFieldIds.includes(fieldId(d)),
+                            }))}
+                            itemComponent={SelectItem}
+                            onChange={(newValue) =>
+                                newValue === null
+                                    ? groupRemove(dimensionId)
+                                    : groupChange(dimensionId, newValue)
+                            }
+                        />
+                    );
+                })}
+
+                <Button
+                    w="fit-content"
+                    size="xs"
+                    leftIcon={<MantineIcon icon={IconPlus} />}
+                    variant="outline"
+                    onClick={groupAdd}
+                    disabled={
+                        groupFieldIds.includes(null) ||
+                        groupFieldIds.length === dimensions.length
+                    }
+                >
+                    Add Group
+                </Button>
+            </Stack>
+
+            <Select
+                label="Metric"
+                placeholder="Select metric"
+                value={metricId}
+                icon={selectedMetric && <FieldIcon item={selectedMetric} />}
+                itemComponent={SelectItem}
+                data={[...metrics, ...customMetrics, ...tableCalculations].map(
+                    (m) => {
+                        const id = isField(m) ? fieldId(m) : m.name;
```

Comment:
```suggestion
                        const id = isField(m) || m.name;
```
small thing 😉

Reply:
logic is correct as is, if isField then give me field id else give me a name

Reply:
😮‍💨 of course! read it wrong, my bad!